ethpm_types/contract_type.py

```diff
@@ -354,6 +355,19 @@ def fallback(self) -> FallbackABI:
         fallback_abi.contract_type = self
         return fallback_abi
 
+    @property
+    def receive(self) -> ReceiveABI:
+        """
+        The ``receive()`` method of the contract, if it has one. A contract may
+        have 0-1 ``receive()`` methods defined. It gets executed when calling
+        the contract with empty calldata. The method is not allowed any arguments
+        and cannot return anything. This property is always available, even if your
+        contract doesn't have a receive method. If you call this method without
+        defining a ``receive()``, it will likely use the ``fallback()`` method.
+        """
+
+        return ReceiveABI(type="receive", stateMutability="payable")
```

Comment:
I think both this one and `fallback` should return optional, if the ABI is defined it means it will accept arbitrary calls. If neither are defined, arbitrary calls should revert.

Another slight invariant is that if `receive()` is defined then `fallback()` should be `nonpayable` (this is Solidity post-v0.6 behavior). If it is not defined, then `fallback()` can be `payable` (Vyper and Solidity pre-v0.6 does this)

Reply:
Even when you define it though, Solidity does not output any ABI to indicate this. The only way to do know would he to analyze the source content

Reply:
Also another issue im having is that the Fallback method has 2 ABIs (in solidity), but even if is use the alternative ABI with io, it does not show in the ABI - it is always void of inputs or outputs....

Reply:
- [x] Optionals
- [ ] Validate non-payable (Second paragraph)